metecho/api/models.py

```diff
@@ -877,19 +878,9 @@ def get_serialized_representation(self, user):
     # end PushMixin configuration
 
     def queue_delete(self, *, originating_user_id):
-        from .jobs import delete_scratch_org_job
-
-        # If the scratch org has no `last_modified_at`, it did not
-        # successfully complete the initial flow run on Salesforce, and
-        # therefore we don't need to notify of its destruction; this
-        # should only happen when it is destroyed during the initial
-        # flow run.
-        if self.last_modified_at:
-            self.delete_queued_at = timezone.now()
-            self.save()
-            self.notify_changed(originating_user_id=originating_user_id)
-
-        delete_scratch_org_job.delay(self, originating_user_id=originating_user_id)
+        # Formerly a background job, currently synchronous:
+        self.delete()
+        self.finalize_delete(originating_user_id=originating_user_id)
```

Comment:
@wlonk I know you're planning to do more cleanup here. Some of my thoughts:

- When reassigning a role, or removing a user, we're now soft-deleting locally and not deleting on SF.
- But, I think we still want to queue and delete on SF (in a background job) when a user explicitly deletes an org? So I don't think we want to just replace our prior logic for org-deletion -- we're supplementing it.
- Also, before re-assigning and un-deleting an org, we should be sure to check that the org is not expired, or at least handle that case gracefully.

Reply:
My understanding, per the Trello card, is that this new logic is used when _reassigning_ from one user to another ("When reassigning an org..."), or when _removing_ an assigned user ("When removing a user who has an org...").

If the user clicks "Delete Org", I understood that we would still actually delete the org on SF, and it would not be a candidate for re-use. Otherwise, as a user how do I clean up my orgs if I hit my SF scratch org limit? I don't think we want users to have to go off to SF to do that. I think this feature is all about how to handle org _auto_-deletion differently.

Reply:
Ah, I follow. To explain my understanding, I have to make explicit the distinction between local orgs and SF orgs. When a local org has no user assigned to it, we locally soft-delete it and don't touch the SF org. If we then have a use reassigned to the local org, we can reassign the attached SF org. If we explicitly delete a local org, we also bring down the SF org. This is clear..

(So I need to keep the queue-delete logic, for that case!)

Reply:
I think maybe I have implemented this? I am struggling to feel certain.

Reply:
@wlonk I think so! We still need to figure out a good solution to https://github.com/oddbird/Metecho/pull/173#discussion_r442273124